src/test/java/org/broadinstitute/consent/http/db/DAOTestHelper.java

```diff
@@ -615,4 +624,30 @@ protected DataAccessRequest createDarForCollection(User user, Integer collection
         dataAccessRequestDAO.insertDARDatasetRelation(dar.getReferenceId(), dataset.getDataSetId());
         return dataAccessRequestDAO.findByReferenceId(dar.getReferenceId());
     }
+
+    protected FileStorageObject createFileStorageObject() {
+        FileCategory category = List.of(FileCategory.values()).get(new Random().nextInt(FileCategory.values().length));
+        String entityId = RandomStringUtils.randomAlphabetic(10);
+
+        return createFileStorageObject(entityId, category);
+    }
+
+    protected FileStorageObject createFileStorageObject(String entityId, FileCategory category) {
+        String fileName = RandomStringUtils.randomAlphabetic(10);
+        String bucketName = RandomStringUtils.randomAlphabetic(10);
+        String gcsFileUri = RandomStringUtils.randomAlphabetic(10);
+        User createUser = createUser();
+        Instant createDate = Instant.now();
+
+        Integer newFileStorageObjectId = fileStorageObjectDAO.insertNewFile(
+                fileName,
+                category.getValue(),
+                bucketName,
+                gcsFileUri,
+                entityId,
+                createUser.getUserId(),
+                createDate
+        );
+        return fileStorageObjectDAO.findFileById(newFileStorageObjectId);
+    }
```

Comment:
This is a pattern I want to avoid. We should move all of this code to the test class that needs it. The anti-pattern we've fallen into here is that this helper class has rapidly expanded to become an uber-service class. I've been removing as much creation code as possible from this class as I get to it.

Reply:
Understood - I moved it back because I started using the `createFileStorage` function in multiple DAO test classes. Where is the correct place for the code, given that it now is used across DAO classes? Is it recommended to copy the code to both DAO classes?

Reply:
They should be local to the test class that uses them, so for example, `FileStorageObjectDAOTest` and in `DatasetDAOTest`. I'm not concerned about DRY in tests, and in fact, DRY in test code can often lead to flaky tests. For example, if I ever needed to make a change to this in `DAOTestHelper` for some other test class, then I would be side-effecting other unit tests which is problematic. The real point of `DAOTestHelper` is to set up the application for startup and shutdown, but that's where it should end, really.